### ISSUE TITLE
Paraformer model finetune OOM

### ISSUE THREAD

**Issue by Joey-xx** [modelscope/FunASR#1828]
Notice: In order to resolve issues more efficiently, please raise issue following the template.
（注意：为了更加高效率解决您遇到的问题，请按照模板提问，补充细节）
#### What is your question?
background：
I changed the tokens.json and seg_dict of speech_paraformer-large_asr_nat-zh-cn-16k-common-vocab8404-pytorch
The data is other language matches the modified tokens.json and seg_dict

question:
when training, the GPU memory increased from 17G to 54G 
here is part of log：

[2024-06-18 18:15:19,010][root][INFO] - val, rank: 0, epoch: 0/50, data_slice: 0/1, step_in_slice: 10/1, step_in_epoch: 0, total step: 5190, (loss_avg_rank: 0.572), (loss_avg_slice: 0.633), (ppl_avg_slice: 1.884e+00), (acc_avg_slice: 0.363), (lr: 0.000e+00), [('loss_att', 0.528), ('acc', 0.354), ('loss_pre', 0.044), ('loss', 0.572), ('batch_size', 24)], {'data_load': '0.000', 'forward_time': '0.067'}, GPU, memory: usage: 2.772 GB, peak: 15.379 GB, cache: 17.807 GB, cache_peak: 17.807 GB
[2024-06-18 18:15:20,089][root][INFO] - val, rank: 0, epoch: 0/50, data_slice: 0/1, step_in_slice: 20/1, step_in_epoch: 0, total step: 5190, (loss_avg_rank: 0.510), (loss_avg_slice: 0.608), (ppl_avg_slice: 1.837e+00), (acc_avg_slice: 0.375), (lr: 0.000e+00), [('loss_att', 0.468), ('acc', 0.465), ('loss_pre', 0.042), ('loss', 0.51), ('batch_size', 21)], {'data_load': '0.000', 'forward_time': '0.114'}, GPU, memory: usage: 2.902 GB, peak: 15.379 GB, cache: 17.807 GB, cache_peak: 17.807 GB
[2024-06-18 18:15:21,068][root][INFO] - val, rank: 0, epoch: 0/50, data_slice: 0/1, step_in_slice: 30/1, step_in_epoch: 0, total step: 5190, (loss_avg_rank: 0.578), (loss_avg_slice: 0.592), (ppl_avg_slice: 1.808e+00), (acc_avg_slice: 0.388), (lr: 0.000e+00), [('loss_att', 0.537), ('acc', 0.487), ('loss_pre', 0.041), ('loss', 0.578), ('batch_size', 15)], {'data_load': '0.000', 'forward_time': '0.089'}, GPU, memory: usage: 3.035 GB, peak: 15.379 GB, cache: 17.807 GB, cache_peak: 17.807 GB
[2024-06-18 18:15:21,920][root][INFO] - val, rank: 0, epoch: 0/50, data_slice: 0/1, step_in_slice: 40/1, step_in_epoch: 0, total step: 5190, (loss_avg_rank: 0.572), (loss_avg_slice: 0.593), (ppl_avg_slice: 1.810e+00), (acc_avg_slice: 0.385), (lr: 0.000e+00), [('loss_att', 0.524), ('acc', 0.413), ('loss_pre', 0.049), ('loss', 0.572), ('batch_size', 21)], {'data_load': '0.000', 'forward_time': '0.080'}, GPU, memory: usage: 2.916 GB, peak: 15.379 GB, cache: 17.807 GB, cache_peak: 17.807 GB
[2024-06-18 18:15:22,731][root][INFO] - val, rank: 0, epoch: 0/50, data_slice: 0/1, step_in_slice: 50/1, step_in_epoch: 0, total step: 5190, (loss_avg_rank: 0.477), (loss_avg_slice: 0.600), (ppl_avg_slice: 1.822e+00), (acc_avg_slice: 0.379), (lr: 0.000e+00), [('loss_att', 0.437), ('acc', 0.322), ('loss_pre', 0.04), ('loss', 0.477), ('batch_size', 14)], {'data_load': '0.000', 'forward_time': '0.080'}, GPU, memory: usage: 3.111 GB, peak: 15.379 GB, cache: 17.807 GB, cache_peak: 17.807 GB
[2024-06-18 18:15:23,514][root][INFO] - val, rank: 0, epoch: 0/50, data_slice: 0/1, step_in_slice: 60/1, step_in_epoch: 0, total step: 5190, (loss_avg_rank: 0.734), (loss_avg_slice: 0.603), (ppl_avg_slice: 1.828e+00), (acc_avg_slice: 0.387), (lr: 0.000e+00), [('loss_att', 0.67), ('acc', 0.451), ('loss_pre', 0.063), ('loss', 0.734), ('batch_size', 17)], {'data_load': '0.000', 'forward_time': '0.075'}, GPU, memory: usage: 2.959 GB, peak: 15.379 GB, cache: 17.807 GB, cache_peak: 17.807 GB
[2024-06-18 18:15:24,303][root][INFO] - val, rank: 0, epoch: 0/50, data_slice: 0/1, step_in_slice: 70/1, step_in_epoch: 0, total step: 5190, (loss_avg_rank: 0.561), (loss_avg_slice: 0.600), (ppl_avg_slice: 1.822e+00), (acc_avg_slice: 0.385), (lr: 0.000e+00), [('loss_att', 0.503), ('acc', 0.462), ('loss_pre', 0.059), ('loss', 0.561), ('batch_size', 21)], {'data_load': '0.000', 'forward_time': '0.077'}, GPU, memory: usage: 2.878 GB, peak: 15.379 GB, cache: 17.807 GB, cache_peak: 17.807 GB
[2024-06-18 18:15:25,084][root][INFO] - val, rank: 0, epoch: 0/50, data_slice: 0/1, step_in_slice: 80/1, step_in_epoch: 0, total step: 5190, (loss_avg_rank: 0.589), (loss_avg_slice: 0.596), (ppl_avg_slice: 1.816e+00), (acc_avg_slice: 0.384), (lr: 0.000e+00), [('loss_att', 0.535), ('acc', 0.403), ('loss_pre', 0.055), ('loss', 0.589), ('batch_size', 24)], {'data_load': '0.000', 'forward_time': '0.079'}, GPU, memory: usage: 2.818 GB, peak: 15.379 GB, cache: 17.807 GB, cache_peak: 17.807 GB
[2024-06-18 18:15:25,887][root][INFO] - val, rank: 0, epoch: 0/50, data_slice: 0/1, step_in_slice: 90/1, step_in_epoch: 0, total step: 5190, (loss_avg_rank: 0.583), (loss_avg_slice: 0.596), (ppl_avg_slice: 1.814e+00), (acc_avg_slice: 0.383), (lr: 0.000e+00), [('loss_att', 0.543), ('acc', 0.367), ('loss_pre', 0.04), ('loss', 0.583), ('batch_size', 16)], {'data_load': '0.000', 'forward_time': '0.079'}, GPU, memory: usage: 2.993 GB, peak: 15.379 GB, cache: 17.807 GB, cache_peak: 17.807 GB
[2024-06-18 18:15:26,483][root][INFO] - Save checkpoint: 0, rank: 0

[2024-06-18 18:15:28,849][root][INFO] -
Checkpoint saved to ./outputs/model.pt.ep0.5190

[2024-06-18 18:15:32,558][root][INFO] - No improvement in acc: 0.3851 < 0.3864, ./outputs/model.pt.ep0.5160
[2024-06-18 18:15:32,559][root][INFO] - Delete: ./outputs/model.pt.ep0.4470
[2024-06-18 18:15:36,527][root][INFO] - train, rank: 0, epoch: 0/50, data_slice: 0/1, step_in_slice: 5200/1, step_in_epoch: 5200, total step: 5200, (loss_avg_rank: 1.519), (loss_avg_slice: 2.228), (ppl_avg_slice: 9.284e+00), (acc_avg_slice: 0.183), (lr: 3.467e-05), [('loss_att', 0.883), ('acc', 0.206), ('loss_pre', 0.636), ('loss', 1.519), ('batch_size', 23)], {'data_load': '0.000', 'forward_time': '1.154', 'backward_and_AllReaduce_time': '0.302', 'optim_time': '0.050', 'total_time': '1.508'}, GPU, memory: usage: 4.236 GB, peak: 46.483 GB, cache: 54.176 GB, cache_peak: 54.176 GB
[2024-06-18 18:15:38,016][root][INFO] - Validate epoch: 0, rank: 0

[2024-06-18 18:15:39,248][root][INFO] - val, rank: 0, epoch: 0/50, data_slice: 0/1, step_in_slice: 10/1, step_in_epoch: 0, total step: 5205, (loss_avg_rank: 0.526), (loss_avg_slice: 0.616), (ppl_avg_slice: 1.852e+00), (acc_avg_slice: 0.359), (lr: 0.000e+00), [('loss_att', 0.483), ('acc', 0.337), ('loss_pre', 0.043), ('loss', 0.526), ('batch_size', 24)], {'data_load': '0.000', 'forward_time': '0.072'}, GPU, memory: usage: 2.772 GB, peak: 46.483 GB, cache: 54.176 GB, cache_peak: 54.176 GB

#### Code

use train_from_local.sh, just changed the GPU number

#### What have you tried?
when OOM， I check the data， there are 259 audio longer than 15s, total 123577, about 200 hours

#### What's your environment?

 - OS (e.g., Linux):Ubuntu 20.04
 - FunASR Version (e.g., 1.0.0):1.0.27
 - ModelScope Version (e.g., 1.11.0):1.15.0
 - PyTorch Version (e.g., 2.0.0):2.2.0
 - How you installed funasr (`pip`, source):source
 - Python version:3.12.3
 - GPU (e.g., V100M32):A100
 - CUDA/cuDNN version (e.g., cuda11.7):nvidia-smi cuda 12.3 but print(torch.version.cuda) = 12.1/cuDNN print(torch.backends.cudnn.version()) = 8902
 - Docker version (e.g., funasr-runtime-sdk-cpu-0.4.1):I use ubuntu 20.04 image and install funasr in container by source
 - Any other relevant information:

**Comment by Joey-xx** [comment]
[Uploading log.txt…]()

**Comment by Joey-xx** [comment]
[log.txt](https://github.com/user-attachments/files/15894739/log.txt)

**Comment by LauraGPT** [comment]
I have update finetune.sh, please update funasr by: `git pull`

https://github.com/modelscope/FunASR/blob/main/examples/industrial_data_pretraining/paraformer/finetune.sh
